notebooks/hfdemo/tinytimemixer/full_benchmarking/gift_leaderboard/src/ttm_gluonts_predictor.py

```diff
@@ -0,0 +1,1005 @@
+            num_prediction_channels (int, optional): Number of prediction channels. Defaults to None.
+            use_mask (bool, optional): Use observed mask. Defaults to False.
+            freq (str, optional): Frequency. Defaults to None.
+            use_valid_from_train (bool, optional): Use unused training data for validation (from 100-x% fewshot).
+                Defaults to True.
+            insample_forecast (bool, optional): Calculate insample statistics for quantile forecasts. Defaults to True.
+            insample_use_train (bool, optional): Use training error statistics along with validation error statistics
+                for insample statistics calculation. Defaults to False.
+            plot_predictions (bool, optional): Plot predictions or not. Defaults to False.
+        """
+        self.prediction_length = prediction_length
+        self.test_data_label = test_data_label
+        self.scale = scale
+        self.scaler = None
+        self.random_seed = random_seed
+        self.term = term
+        self.ds_name = ds_name
+        self.out_dir = out_dir
+        self.upper_bound_fewshot_samples = upper_bound_fewshot_samples
+        self.force_short_context = force_short_context
+        self.min_context_mult = min_context_mult
+        self.past_feat_dynamic_real_exist = past_feat_dynamic_real_exist
+        self.num_prediction_channels = num_prediction_channels
+        self.freq = freq
+        self.use_mask = use_mask
+        self.use_valid_from_train = use_valid_from_train
+        self.insample_forecast = insample_forecast
+        self.insample_use_train = insample_use_train
+        self.plot_predictions = plot_predictions
+        self.quantile_keys = [
+            "0.1",
+            "0.2",
+            "0.3",
+            "0.4",
+            "0.5",
+            "0.6",
+            "0.7",
+            "0.8",
+            "0.9",
+            "mean",
+        ]
+
+        self.insample_errors = None
+
+        if force_short_context:
+            logger.info(f"Forcing short context: H = {prediction_length}, CL={prediction_length*min_context_mult}")
+
+        self.device = "cuda" if torch.cuda.is_available() else "cpu"
+
+        if "dropout" in kwargs and kwargs["dropout"] is None:
+            del kwargs["dropout"]
+        if "head_dropout" in kwargs and kwargs["head_dropout"] is None:
+            del kwargs["head_dropout"]
+        if "loss" in kwargs and kwargs["loss"] is not None:
+            self.loss = kwargs["loss"]
+            if kwargs["loss"] not in ["mse", "mae", "huber", "pinball"]:
+                del kwargs["loss"]
+        if "prediction_channel_indices" in kwargs:
+            self.prediction_channel_indices = kwargs["prediction_channel_indices"]
+
+        # Get model
+        self._get_gift_model(model_path, context_length, **kwargs)
+
+    def _get_gift_model(self, model_path: str, context_length: int, **kwargs):
```

Comment:
Maybe this can just be called `_get_model` -- then we should add some comment at the top level stating that this is designed for the current gift-eval benchmark?